terraform/services/api-rds/data.tf

```diff
@@ -1,54 +1,124 @@
 locals {
   secret_date = "2020-01-02-09-15-01"
+  gedit_security_group_names = [
+    "bcda-${var.env}-vpn-private",
+    "bcda-${var.env}-vpn-public",
+    "bcda-${var.env}-remote-management",
+    "bcda-${var.env}-enterprise-tools"
+  ]
 }
 
 data "aws_default_tags" "data_tags" {}
 
+# Secrets for "ab2d" app
 data "aws_secretsmanager_secret" "secret_database_password" {
-  name = "ab2d/${local.db_name}/module/db/database_password/${local.secret_date}"
+  count = var.app == "ab2d" ? 1 : 0
+  name  = "ab2d/${local.db_name}/module/db/database_password/${local.secret_date}"
 }
+
 data "aws_secretsmanager_secret_version" "database_password" {
-  secret_id = data.aws_secretsmanager_secret.secret_database_password.id
+  count     = var.app == "ab2d" ? 1 : 0
+  secret_id = data.aws_secretsmanager_secret.secret_database_password[0].id
 }
 
 data "aws_secretsmanager_secret" "secret_database_user" {
-  name = "ab2d/${local.db_name}/module/db/database_user/${local.secret_date}"
+  count = var.app == "ab2d" ? 1 : 0
+  name  = "ab2d/${local.db_name}/module/db/database_user/${local.secret_date}"
 }
+
 data "aws_secretsmanager_secret_version" "database_user" {
-  secret_id = data.aws_secretsmanager_secret.secret_database_user.id
+  count     = var.app == "ab2d" ? 1 : 0
+  secret_id = data.aws_secretsmanager_secret.secret_database_user[0].id
+}
+
+# Secrets for "bcda" app
+data "aws_secretsmanager_secret" "database_secret" {
+  count = var.app == "bcda" ? 1 : 0
+  name  = "${var.app}/${var.env}/rds-main-credentials"
 }
 
+data "aws_secretsmanager_secret_version" "database_secret_version" {
+  count     = var.app == "bcda" ? 1 : 0
+  secret_id = data.aws_secretsmanager_secret.database_secret[0].id
+}
+
+
 data "aws_caller_identity" "current" {}
 
 data "aws_region" "current" {}
 
 data "aws_vpc" "target_vpc" {
   filter {
-    name   = "tag:Name"
-    values = ["${local.db_name}"]
+    name = "tag:Name"
+    values = [
+      var.app == "ab2d" ? local.db_name : "${var.app}-${var.env}-vpc"
+    ]
   }
 }
 
 data "aws_subnet" "private_subnet_a" {
+  count = var.app == "ab2d" ? 1 : 0
   filter {
     name   = "tag:Name"
     values = ["${local.db_name}-private-a"]
   }
 }
 
 data "aws_subnet" "private_subnet_b" {
+  count = var.app == "ab2d" ? 1 : 0
   filter {
     name   = "tag:Name"
     values = ["${local.db_name}-private-b"]
   }
 }
 
+data "aws_subnets" "bcda_subnets" {
+  count = var.app == "bcda" ? 1 : 0 # Only create this data source if app is 'bcda'
+  filter {
+    name = "tag:Name"
+    values = flatten([
+      var.app == "bcda" && var.env == "opensbx" ? ["${var.app}-${var.env}-az1-data", "${var.app}-${var.env}-az2-data"] : [],
+      var.app == "bcda" && var.env != "opensbx" ? ["${var.app}-${var.env}-az1-data", "${var.app}-${var.env}-az2-data", "${var.app}-${var.env}-az3-data"] : []
+    ])
+  }
+}
```

Comment:
Rather than "count = var.app" on these subnet resources, there should be one `data "aws_subnets" "db"` resource that has a ternary on the filter values for each app. Also, we can drop the "az3-data" subnet from bcda. The only database in that us-east-1c subnet is dev and it can be restored.

Reply:
Looking better. The subnet naming standard is the same for bcda and dpc so the values could look like this:

```
    values = var.app == "ab2d" ? [
        "${local.db_name}-private-a",
        "${local.db_name}-private-b"
      ] : [
        "${var.app}-${var.env}-az1-data",
        "${var.app}-${var.env}-az2-data"
      ]
```